magefile.go

```diff
@@ -181,3 +196,147 @@ func licenser(mode licenserMode) error {
 
 	return nil
 }
+
+type Benchmark mg.Namespace
+
+func InstallBenchStat() error {
+	err := gotool.Install(gotool.Install.Package(goBenchstats))
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (Benchmark) Run(ctx context.Context, outputFile string) error {
+	mg.Deps(InstallBenchStat)
+	fmt.Println(">> go benchmark:", "Testing")
+	args := []string{
+		"test",
+		fmt.Sprintf("-count=%d", benchmarkCount),
+		"-bench=.",
+		"-run=Bench#",
+	}
+	for _, pkg := range goBenchmarkPackages {
+		args = append(args, filepath.Join("github.com/elastic/elastic-agent-shipper-client", pkg, "..."))
+	}
+	goTestBench := makeCommand(ctx, nil, "go", args...)
+
+	// Wire up the outputs.
+	var outputs []io.Writer
+	if outputFile != "" {
+		fileOutput, err := os.Create(createDir(outputFile))
+		if err != nil {
+			return errors.Wrap(err, "failed to create go test output file")
+		}
+		defer fileOutput.Close()
+		outputs = append(outputs, fileOutput)
+	}
+	output := io.MultiWriter(outputs...)
+	goTestBench.Stdout = io.MultiWriter(output, os.Stdout)
+	goTestBench.Stderr = io.MultiWriter(output, os.Stderr)
+
+	err := goTestBench.Run()
+
+	var goTestErr *exec.ExitError
+	if err != nil {
```

Comment:
I'm not sure why the error checks are in two separate if statements, it's a tad hard to follow at first. It might be a little more concise if we just do something like `if errors.Is() {...} else if errors.Is() {...} else err != nil {...}` or something like that.